config/config.go

```diff
@@ -58,6 +58,8 @@ type Config struct {
 	SourceBlockchains      []*SourceBlockchain      `mapstructure:"source-blockchains" json:"source-blockchains"`
 	DestinationBlockchains []*DestinationBlockchain `mapstructure:"destination-blockchains" json:"destination-blockchains"`
 	ProcessMissedBlocks    bool                     `mapstructure:"process-missed-blocks" json:"process-missed-blocks"`
+	DeciderHost            string                   `mapstructure:"decider-host" json:"decider-host"`
+	DeciderPort            *uint16                  `mapstructure:"decider-port" json:"decider-port"`
```

Comment:
Just calling this out because I wasn't aware how it worked exactly myself:

AvalancheGo communicates with VM plugins via gprc, but instead of assuming the VMs are running and providing their host/port in the configuration, it starts the configured VMs itself on the first port available on `127.0.0.1` [here](https://github.com/ava-labs/avalanchego/blob/master/vms/rpcchainvm/vm_client.go#L183).

That type of pattern may be less error prone for relayers to set up. i.e: the relayer config would specify the decider plugin(s) (if any) to run for a given source chain, and the application would handle initializing/configuring them itself.

Not suggesting we do this now, but could be worth considering going forward.

Reply:
I think that what Michael described is the generally preferred pattern for colocated services, and I agree that less config is better and less error prone. 

A counter example for why we would want to keep it, is that this allows for using remote deciders.

Reply:
Cam and I also discussed this. For this iteration I was going for the simplest thing possible, and sub-process management proved non-trivial, so I went with this host/port approach instead, since we said that ultimately we want to be able to support both local and remote deciders.

Going forward, I'm open to doing the sub-process/plugin approach; we _may_ be able to leverage the sub-process modules available in avalanchego, but if not then the complexity added to the code base for this would be a lot, at least as compared to the host/port approach.

Also, at the moment I'm leaning towards just the host/port approach so that there's one single config interface, rather than one for a local decider (the plugin path) and a separate/different one for a remote decider (the host/port).